include/rocksdb/db.h

```diff
@@ -216,6 +216,9 @@ class DB {
   // close column family instead of deleting column family handle directly
   virtual Status DestroyColumnFamilyHandle(ColumnFamilyHandle* column_family);
 
+  // Same as above, should called without mutex held and not on write thread.
+  virtual ColumnFamilyHandle* GetColumnFamilyHandleUnlocked(uint32_t column_family_id) = 0;
```

Comment:
Is this new API needed? I think the user should hold column family handle and pass it back to the db. How does the rest of Java binding handle column family handle?

Reply:
@yiwu-arbug The Java API before this only knew of `ColumnFamilyDescriptor` and `ColumnFamilyHandle`, so I have tried to keep it that way for users.

Working with Column Family ids is a little tricky, and I did not want to expose the end-user to that.

I could perhaps keep a map of `id` -> `column family handle` in the `RocksDB` object, but then due to multi-threading I would need to introduce a mutex or some sort of concurrent map with CoW; such an approach seemed perhaps more complex than just using what appeared to already be present in `DBImpl`.

I am open to ideas and happy to discuss further...

Reply:
So my question is why we need to deal with column_family_id instead of ColumnFamilyHandle if we don't expose it to user? I'm wanting to avoid expose this method at least in the public API (include/rocksdb/*.h).

Reply:
I agree with @yiwu-arbug here. We have been consciously trying not to expose additional APIs and options to the end users unless absolutely necessary, as it is quite difficult to deprecate any of them later. 
It would be good to see if we can handle this code internally, without exposing a new end user API.

Reply:
Okay sounds good. What do you suggest then?

Reply:
@adamretter I don't have a good suggestion here, but I don't feel a strong need to expose this in the C++ public API. 
Since the RocksDB JNI code lives alongside RocksDB C++ implementation code, I was wondering if JNI itself could access RocksDB's internal implementation headers? May be not the cleanest way, but just tying to think through all the options that we have here.